### DISCUSSION TITLE
v3.5.0-M3

### DISCUSSION THREAD

**Discussion by nafg** [slick/slick#2711] [Announcements]
### Highlights 🎉
* Macroless b interpolator by @nafg in https://github.com/slick/slick/pull/2684
* Remove HList features by @nafg in https://github.com/slick/slick/pull/2693
* Remove specialization by @nafg in https://github.com/slick/slick/pull/2695
* Remove auto-mapping by @nafg in https://github.com/slick/slick/pull/2698
* Remove macros from Plain SQL interpolators by @nafg in https://github.com/slick/slick/pull/2696
* un-deprecate `countDistinct` method by @xuwei-k in https://github.com/slick/slick/pull/2649
* Remove deprecated features by @nafg in https://github.com/slick/slick/pull/2710
### Bugs fixed 🐛️
* avoid deprecated `java.net.URL` constructor by @xuwei-k in https://github.com/slick/slick/pull/2691
* OracleProfile: restore parentheses by @nafg in https://github.com/slick/slick/pull/2697
* For some reason searching by dates in some timezones occasionally fails on Oracle 11g XE, avoid them in tests by @OlegYch in https://github.com/slick/slick/pull/2685
### Chores 🛠
* Update dependency org.xerial.sbt:sbt-sonatype to v3.9.18 by @renovate in https://github.com/slick/slick/pull/2675
* Update logback-classic to 1.3.6 by @scala-steward in https://github.com/slick/slick/pull/2677
* Update dependency org.slf4j:slf4j-api to v2.0.7 by @renovate in https://github.com/slick/slick/pull/2679
* Update dependency org.postgresql:postgresql to v42.6.0 by @renovate in https://github.com/slick/slick/pull/2680
* Avoid compiling entire project on Intellij import by @OlegYch in https://github.com/slick/slick/pull/2678
* Update dependency com.typesafe:sbt-mima-plugin to v1.1.2 by @renovate in https://github.com/slick/slick/pull/2699
* Update sbt-sdlc to 0.3.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2704
* Update dependency mysql:mysql-connector-java to v8.0.33 by @renovate in https://github.com/slick/slick/pull/2706
* Update logback-classic to 1.3.7 by @slick-scala-steward in https://github.com/slick/slick/pull/2707
### Other Changes
* Some cleanups by @nafg in https://github.com/slick/slick/pull/2683
* An indentation fix by @nafg in https://github.com/slick/slick/pull/2692
* Line breaks after infix operator by @nafg in https://github.com/slick/slick/pull/2694
* OracleProfile: Delete unused and incomplete OracleLocalTimeJdbcType from #1872 by @nafg in https://github.com/slick/slick/pull/2686

## New Contributors
* @OlegYch made their first contribution in https://github.com/slick/slick/pull/2678

**Full Changelog**: https://github.com/slick/slick/compare/v3.5.0-M2...v3.5.0-M3

<hr /><em>This discussion was created from the release <a href='https://github.com/slick/slick/releases/tag/v3.5.0-M3'>v3.5.0-M3</a>.</em>

**Comment by xuwei-k** [comment]
😢 

https://github.com/slick/slick/actions/runs/4784700476/jobs/8530483674


> [error] (slick / signedArtifacts) Failure running 'gpg --batch --pinentry-mode loopback --passphrase *** --detach-sign --armor --use-agent --output /home/runner/work/slick/slick/slick/target/scala-2.13/slick_2.13-3.5.0-M3.pom.asc /home/runner/work/slick/slick/slick/target/scala-2.13/slick_2.13-3.5.0-M3.pom'.  Exit code: 2
> [error] (codegen / signedArtifacts) Failure running 'gpg --batch --pinentry-mode loopback --passphrase *** --detach-sign --armor --use-agent --output /home/runner/work/slick/slick/slick-codegen/target/scala-2.13/slick-codegen_2.13-3.5.0-M3.pom.asc /home/runner/work/slick/slick/slick-codegen/target/scala-2.13/slick-codegen_2.13-3.5.0-M3.pom'.  Exit code: 2
> [error] (testkit / signedArtifacts) Failure running 'gpg --batch --pinentry-mode loopback --passphrase *** --detach-sign --armor --use-agent --output /home/runner/work/slick/slick/slick-testkit/target/scala-2.13/slick-testkit_2.13-3.5.0-M3.pom.asc /home/runner/work/slick/slick/slick-testkit/target/scala-2.13/slick-testkit_2.13-3.5.0-M3.pom'.  Exit code: 2
> [error] (hikaricp / signedArtifacts) Failure running 'gpg --batch --pinentry-mode loopback --passphrase *** --detach-sign --armor --use-agent --output /home/runner/work/slick/slick/slick-hikaricp/target/scala-2.13/slick-hikaricp_2.13-3.5.0-M3.pom.asc /home/runner/work/slick/slick/slick-hikaricp/target/scala-2.13/slick-hikaricp_2.13-3.5.0-M3.pom'.  Exit code: 2
>

**Comment by nafg** [comment]
I know. I think it was because the key expired, but I think I updated it and it's still not working. I'm trying to reproduce elsewhere, but it's possible I'll have to push a change, in which case I may delete the tag and recreate it.

**Comment by nafg** [comment]
So, it seems at first the issue was the key expired, but then I updated the expiration date and retried, but then the issue was that I exported it the wrong way.

I looked into it more, exported it again and updated the github secrets, and reran the publish job, and it worked!

So now it is indeed [on maven central](https://repo1.maven.org/maven2/com/typesafe/slick/slick_2.13/3.5.0-M3/).

**Comment by xuwei-k** [comment]
All tests passed in my production code with slick 3.5.0-M3. thank you for your works!
